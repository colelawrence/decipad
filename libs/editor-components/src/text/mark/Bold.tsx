import { atoms } from '@decipad/ui';
import { PlateComponent } from '@decipad/editor-types';

export const Bold: PlateComponent = ({ attributes, children }) => {
  return (
    <span {...attributes}>
      <atoms.Bold>{children}</atoms.Bold>
    </span>
  );
};
