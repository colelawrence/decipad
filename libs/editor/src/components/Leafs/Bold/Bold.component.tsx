import styled from '@emotion/styled';
import { SlatePluginComponent } from '@udecode/slate-plugins';
import React from 'react';

const Leaf = styled('span')({
  fontWeight: 'bold',
});

export const Bold: SlatePluginComponent = ({ attributes, children }) => {
  return <Leaf {...attributes}>{children}</Leaf>;
};
