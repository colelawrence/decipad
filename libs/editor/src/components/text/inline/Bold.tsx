import { atoms } from '@decipad/ui';
import { PlatePluginComponent } from '@udecode/plate';

export const Bold: PlatePluginComponent = ({ attributes, children }) => {
  if (!('data-slate-leaf' in attributes)) {
    throw new Error('Bold is a leaf');
  }

  return <atoms.Bold slateAttrs={attributes}>{children}</atoms.Bold>;
};
