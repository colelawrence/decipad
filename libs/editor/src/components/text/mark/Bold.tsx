import { atoms } from '@decipad/ui';
import { PlateComponent } from '../../../types';

export const Bold: PlateComponent = ({ attributes, children }) => {
  return (
    <span {...attributes}>
      <atoms.Bold>{children}</atoms.Bold>
    </span>
  );
};
