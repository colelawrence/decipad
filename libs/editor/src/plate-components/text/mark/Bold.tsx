import { types } from '@decipad/editor-config';
import { atoms } from '@decipad/ui';

export const Bold: types.PlateComponent = ({ attributes, children }) => {
  return (
    <span {...attributes}>
      <atoms.Bold>{children}</atoms.Bold>
    </span>
  );
};
