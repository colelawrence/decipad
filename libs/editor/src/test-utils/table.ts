import {
  ELEMENT_TABLE,
  ELEMENT_TH,
  ELEMENT_TR,
  ELEMENT_TD,
} from '@udecode/plate';
import {
  InteractiveTable,
  TableCellType,
  Td,
  Th,
} from '../plugins/InteractiveTable/table';
import {
  ELEMENT_TABLE_CAPTION,
  ELEMENT_THEAD,
  ELEMENT_HEAD_TR,
  ELEMENT_TBODY,
} from '../utils/elementTypes';

/**
 * Creates a row with 1 named column and {@param numberOfRows} rows of non-empty cells.
 */
export const createTable = (numberOfRows: number): InteractiveTable => ({
  type: ELEMENT_TABLE,
  children: [
    {
      type: ELEMENT_TABLE_CAPTION,
      children: [{ text: 'Table' }],
    },
    {
      type: ELEMENT_THEAD,
      children: [
        {
          type: ELEMENT_HEAD_TR,
          children: [
            {
              type: ELEMENT_TH,
              attributes: { cellType: 'string' },
              children: [{ text: 'Col 1' }],
            },
          ],
        },
      ],
    },
    {
      type: ELEMENT_TBODY,
      children: Array.from({ length: numberOfRows }, (_, i) => ({
        type: ELEMENT_TR,
        children: [
          {
            type: ELEMENT_TD,
            attributes: { cellType: 'string' },
            children: [{ text: `Col 1 Row ${i + 1}` }],
          },
        ],
      })),
    },
  ],
});

export const createTableWithTypeAndData = (
  cellType: TableCellType,
  data: string[]
): InteractiveTable => ({
  type: ELEMENT_TABLE,
  children: [
    {
      type: ELEMENT_TABLE_CAPTION,
      children: [{ text: 'Table' }],
    },
    {
      type: ELEMENT_THEAD,
      children: [
        {
          type: ELEMENT_HEAD_TR,
          children: [
            {
              type: ELEMENT_TH,
              attributes: { cellType },
              children: [{ text: 'Col 1' }],
            },
          ],
        },
      ],
    },
    {
      type: ELEMENT_TBODY,
      children: data.map((text) => ({
        type: ELEMENT_TR,
        children: [
          {
            type: ELEMENT_TD,
            attributes: { cellType },
            children: [{ text }],
          },
        ],
      })),
    },
  ],
});

export const printTable = ({
  children: [_, thead, tbody],
}: InteractiveTable): string[] => {
  const tdsToStrings = (tds: (Th | Td)[]) =>
    tds.map((td) => td.children[0].text).join(' | ');

  return [
    tdsToStrings(thead.children[0].children),
    ...tbody.children.map((tr) => tdsToStrings(tr.children)),
  ];
};
