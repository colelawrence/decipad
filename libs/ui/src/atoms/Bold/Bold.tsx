import { css } from '@emotion/react';
import { ReactNode } from 'react';
import { p16Bold } from '../../primitives';
import { SlateLeafProps } from '../../utils';

const styles = css(p16Bold);

interface BoldProps extends SlateLeafProps {
  readonly children: ReactNode;
}
export const Bold = ({ children }: BoldProps): ReturnType<React.FC> => {
  return <span css={styles}>{children}</span>;
};
