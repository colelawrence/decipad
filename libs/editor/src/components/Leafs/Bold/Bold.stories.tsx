import { Bold } from './Bold.component';

export default {
  title: 'Editor/Leafs/Bold',
  component: Bold,
};

export const Default = (args: { text: string }) => (
  <Bold
    leaf={{ attributes: {}, text: '' }}
    element={{ children: [{ text: '' }] }}
    text={{ text: '' }}
    attributes={{
      'data-slate-node': 'element',
      ref: null,
      'data-slate-leaf': true,
    }}
  >
    {args.text}
  </Bold>
);

Default.args = {
  text: 'Bold',
};

Default.storyName = 'Bold';
