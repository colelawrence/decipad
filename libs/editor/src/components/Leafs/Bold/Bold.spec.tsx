import React from 'react';
import { render } from 'test-utils';
import { Default } from './Bold.stories';

describe('Bold Leaf', () => {
  it('matches snapshot', () => {
    const { container } = render(<Default text="Bold" />);
    expect(container).toMatchSnapshot();
  });
});
