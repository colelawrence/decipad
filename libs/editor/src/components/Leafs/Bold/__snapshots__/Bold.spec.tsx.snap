// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bold Leaf matches snapshot 1`] = `
.emotion-0 {
  font-weight: bold;
}

<div>
  <span
    class="emotion-0"
    data-slate-leaf="true"
    data-slate-node="element"
  >
    Bold
  </span>
</div>
`;
